### ISSUE TITLE
mingw after refactor-build: misc issues, part II

### ISSUE THREAD

**Issue by sav-ix** [openssl/openssl#711]
This is a continuation of previous [mingw after refactor-build: misc issues](https://github.com/openssl/openssl/issues/625) thread, since it became too long. Digest of pending questions, mentioned in it and has not continued:

1. @richsalz [request1](https://github.com/openssl/openssl/issues/625#issuecomment-180059693), [request2](https://github.com/openssl/openssl/issues/625#issuecomment-180179672), and related @vszakats [review](https://github.com/openssl/openssl/issues/625#issuecomment-180098101),

2. @levitte, re-draw your attention, that [tip](https://github.com/openssl/openssl/issues/625#issuecomment-180075510) is missing in [Wiki](https://wiki.openssl.org/index.php/Compilation_and_Installation) and `OpenSSL/Install*` manuals. This info could be obvious for OpenSSL developers, but not for users,

3. [request](https://github.com/openssl/openssl/issues/625#issuecomment-180262776), and related @vszakats [review1](https://github.com/openssl/openssl/issues/625#issuecomment-180326838), [review2](https://github.com/openssl/openssl/issues/625#issuecomment-180350871),

4. with one of the [patches](https://github.com/openssl/openssl/issues/625#issuecomment-182932360) above `-march=` parameter was removed from `mingw` configuration, but still exists in few other [x86 configurations](https://github.com/openssl/openssl/issues/625#issuecomment-181154837); could you clarify, for what purposes they were keeped on listed platforms?

5. regardless of whether MSYS2/PERL [flaws](https://github.com/Alexpux/MSYS2-packages/issues/445) would be fixed or not, IMO it still remains the best solution for OpenSSL builds using MinGW* on Windows; thus it make sence:
-- to not add in `README.PERL` [notes](https://github.com/openssl/openssl/issues/625#issuecomment-183399965) concerning MSYS2/PERL flaws (especially while ActiveState and Strawberry Perl [results](https://github.com/openssl/openssl/issues/625#issuecomment-183446370) are much worse for MinGW-W64 builds),
-- add notes to section `GNU C (MinGW/MSYS)` of file `OpenSSL/Install.win`, that compared to other available GNU Shell environments on Windows (GnuWin32, busybox, MSYS, etc.), MSYS2 (msys2.github.io) is the best available solution (final wording should be more correct), and describe in short how to bypass most common errors,

6. applink/uplink related posts:
-- by @dot-asm (https://github.com/openssl/openssl/issues/625#issuecomment-181026632),
-- by @levitte (https://github.com/openssl/openssl/issues/625#issuecomment-183400540),
-- by @vszakats (https://github.com/openssl/openssl/issues/625#issuecomment-183413297),
left more questions than answers; but since this topic acquired attributes of holy war, probably not worth spending time on it,

7. @vszakats [suggestion1](https://github.com/openssl/openssl/issues/625#issuecomment-183675752), [suggestion2](https://github.com/openssl/openssl/issues/625#issuecomment-183676136) concerning additional linker options.

8. make `man` and `html` commands are broken for `unixmake` builds, while no much sense in their fixing, as it would be deprecated in future.

**Comment by sav-ix** [comment]
> Not sure what the "launch folder" is. I know about the source folder and the build folder, is it one of those?

@levitte, consider, that all that definitions like `build folder` not described in `Configurations/Readme`, and known for OpenSSL Developers only. Hence the inaccuracies, as 3rd party persons have to rely only on their intuition, while choosing the definitions.

> I'm sorry, I thought that was a no-brainer. It's simple, don't run several builds at the same time in the same build directory.

Well, for those, who are not familiar with the `unified` scheme implementation, that could be a new. At least until first build failures appeared. Same as above, not all *undocumented* features are obvious to 3rd party persons.

All above was FYI only. It don't mean that is needed to do something.

**Comment by levitte** [comment]
@sav_ix, please be patient. There's been a lot going on with this stuff alone, can't do or remember everything all the time. 

In #625, the focus was entirely and solely on MingW and Cygwin. The other configs will be looked at. Right now, though, I'm taking a break for a few hours.

**Comment by sav-ix** [comment]
> @sav_ix, please be patient

Noone hurry you. Answer, when it will be convenient.

**Comment by levitte** [comment]
Ah, another thing, please avoid this kind of mega combo issue. Split it up into one specific issue per issue. This kind of everything thrown into one is painful to handle, hard as hell to follow, and as you've seen, details are lost on the way.

**Comment by sav-ix** [comment]
Ok. But consider, mine is only issue (3). All other were put here for your convenience, in case you're still interested return to them.

**Comment by dot-asm** [comment]
> Say, me already raised issue (3) on OpenSSL bugtracker

I.e. question about -std=c11. The answer provided by @vszakats is sufficient, i.e. it's not a bug/error. OpenSSL consciously uses various compiler extensions such as inline assembly, and attempt to impose *strict* standard compliance as suggested is bound to fail. Unless you pass -DPENDANTIC along.

So I'm closing this. But before you rush opening more please consider @levitte's comment, as well as keep in mind that our goal is not to accommodate latest-n-greatest, but rather certain common denominator, yet providing enough flexibility for those who want to deploy additional bells-n-whistles (such as fancy linker flags, removal of image base pinning is pending in review).

**Comment by sav-ix** [comment]
> I.e. question about -std=c11.

The [issue](https://rt.openssl.org/Ticket/Display.html?id=4255) contained two different questions. Besides `-std=c11` usage it concerned OpenSSL test suite failures for MinGW builds. And when it [came turn](https://github.com/openssl/openssl/issues/625#issuecomment-186166987) of `make test` failures, you've closed issue without studying details.

> as well as keep in mind that our goal is not to accommodate latest-n-greatest

`-std=C89` is not  latest-n-greatest GCC switch, while it reproduces described error too.

> The answer provided by @vszakats is sufficient, i.e. it's not a bug/error.

I am grateful to @vszakats for his review. But I also hoped to receive an answer from OpenSSL *Developers*. Your answer is eloquent enough, thanks.

> But before you rush opening more please consider @levitte's comment,

I found *nothing* in @levitte's comment, that he not interested in this issue, or something else. How did you - it's just a lack of respect to other people's work. As it cost some time to describe the error and reach this point.
I won't rush @levitte re-open this issue. He can do it, if considers my help useful.

Cheers.

**Comment by dot-asm** [comment]
>>   I.e. question about -std=c11.

> The issue contained two different questions.
> ...
> I found nothing in @levitte's comment,

I was referring to @levitte's suggestion to resist the temptation to throw things together like it happened in 625. It's overly easy for details to get blurred in the heat of action. And it's happening here again. Indeed, what did you think I was referring to, and consider the misunderstanding that it's already causing. For example, as for my reference to latest-n-greatest, it was reference to (7) and the message was  "something will be done about --image-base, but the rest is up to person who invokes Configure". While you seem to have interpreted as "-std=cXX is too fancy". :-) That's exactly why it's more appropriate to *not* pile things up, as @levitte suggested. Bottom line is not all of the 8 bullets would require new issue. And rationale behind closing this one was "since you've kind of chosen (3) as the one defining this thread, then it's considered as non-issue". Well, I kind of disregarded those log files, sorry. But number of things has changed in the process, so it's hard to tell which of the problem you refer in that ticket still persist... Could you reevaluate and report? In RT perhaps...

**Comment by sav-ix** [comment]
> I was referring to @levitte's suggestion to resist the temptation to throw things together like it happened in 625

"Things were thrown together" with 1st post of this thread, before @levitte request to avoid such things. But in other case they could be just forgotten under a heap of other posts in 625. Is that a better option?
And how had I to know, which of that questions still worth the attention of OpenSSL Developers? Open 8 new issues?

> That's exactly why it's more appropriate to not pile things up, as @levitte suggested.

IMO, you have hastened with the closing this topic. At least to wrote a short answer for each question *before* closing, would allow to avoid misunderstanding.

> Could you reevaluate and report?

If none of issues above are interested for OpenSSL Developers, the only opened question for me is successful run of `make test` command. If, for some reasons, it require new issue (instead of this), I gladly join that thread. But for now I will refrain from opening new issues.

**Comment by richsalz** [comment]
@sav-ix, I think you are being overly sensitive.

There is no doubt that the threads here have wound around and around, and two of the dev team members are saying, politely I thought, that it is becoming too hard to follow.  They offered suggestions and a plea for help to make it easier for them to get the right thing done.

Don't stop finding and reporting issues and bugs.

For each one, email to rt@openssl.org and it will get tracked as a separate bug.

Thanks for your help in making OpenSSL better.
